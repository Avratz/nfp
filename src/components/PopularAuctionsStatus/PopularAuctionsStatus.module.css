.auctionStatus {
  background-color: var(--neutral-2);
  border-radius: 1.5rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  padding: 2rem 0;
  gap: 1.5rem;
  box-shadow: 0 4rem 4rem -3rem rgba(31, 47, 70, 0.12);
}

.auctionStatusTitle {
  font-family: var(--font-poppins);
  font-weight: var(--font-medium);
  padding-bottom: 0.25rem;
}

.priceUsd {
  padding-top: 0.25rem;
  color: var(--neutral-4);
}

.timer {
  display: flex;
  gap: 3rem;
  align-items: center;
  justify-content: center;
  padding-top: 0.25rem;
}

.timeValue {
  font-family: var(--font-dmSans);
  font-weight: var(--font-bold);
  font-size: var(--text-4xl);
  line-height: var(--line-4xl);
}

.timeUnit {
  font-family: var(--font-poppins);
  font-weight: var(--font-medium);
  color: var(--neutral-4);
}

.skeleton {
  margin: 0.5rem auto 0;
  background-color: var(--neutral-5);
  padding: 0.75rem 0;
  width: 30%;
  animation: pulse 1.5s infinite;
}

@keyframes pulse {
  0% {
    opacity: 0.5;
  }
  50% {
    opacity: 1;
  }
  100% {
    opacity: 0.5;
  }
}
