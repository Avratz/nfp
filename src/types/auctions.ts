export type Auction = {
  id: number
  instantPrice: string
  highestBid: string
  author: string
  authorAvatar: string
  type: string
  stock: number
  likes: number
  createdAt: Date
  endsAt: Date
  media: {
    id: number
    image: string
    image2x: string
  }
  attributes: {
    id: number
    color: string
    type: string
  }
  bidUsers: [
    {
      id: number
      name: string
      avatar: string
    },
  ]
}
