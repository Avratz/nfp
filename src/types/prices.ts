export type EthereumPrice = {
  eth: number
  usd: number
}
