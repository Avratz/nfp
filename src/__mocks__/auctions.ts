import { Auction } from '@/types/auctions'

export const allAuctions: Auction[] = [
  {
    id: 10,
    instantPrice: '4.45 ETH',
    highestBid: '0.010 ETH',
    author: 'Fernando Medina',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/07_xrihyb.png',
    type: 'Art',
    stock: 1,
    likes: 2,
    createdAt: '2022-06-05T11:42:33.000Z',
    endsAt: '2022-08-06T11:42:33.000Z',
    media: {
      id: 4,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121559/NFPAISANOS/img_125_mrxfbh.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1500/v1657121559/NFPAISANOS/img_125_mrxfbh.png',
    },
    attributes: {
      id: 4,
      color: 'black',
      type: 'epic',
    },
    bidUsers: [
      {
        id: 6,
        name: 'Sofia Coolen',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/05_rdi4su.png',
      },
      {
        id: 7,
        name: 'Martin Acosta',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/04_uunxgl.png',
      },
    ],
  },
  {
    id: 11,
    instantPrice: '2.45 ETH',
    highestBid: '0.100 ETH',
    author: 'Martin Acosta',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/06_mjddoi.png',
    type: 'Art',
    stock: 3,
    likes: 10,
    createdAt: '2022-06-06T11:42:33.000Z',
    endsAt: '2022-08-06T11:42:33.000Z',
    media: {
      id: 5,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121554/NFPAISANOS/img_131_d7wmyk.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1500/v1657121559/NFPAISANOS/img_125_mrxfbh.png',
    },
    attributes: {
      id: 5,
      color: 'pink',
      type: 'rare',
    },
    bidUsers: [
      {
        id: 8,
        name: 'Nicolas Sieiro',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/02_nqwu5e.png',
      },
      {
        id: 9,
        name: 'Carla Luna',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/03_ifygi0.png',
      },
    ],
  },
  {
    id: 12,
    instantPrice: '2.15 ETH',
    highestBid: '0.500 ETH',
    author: 'Santiago Lopez',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/04_uunxgl.png',
    type: 'Photography',
    stock: 5,
    likes: 10,
    createdAt: '2022-06-07T11:42:33.000Z',
    endsAt: '2022-08-06T11:42:33.000Z',
    media: {
      id: 6,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121551/NFPAISANOS/img_126_ae6zew.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121551/NFPAISANOS/img_126_ae6zew.png',
    },
    attributes: {
      id: 6,
      color: 'orange',
      type: 'uncommon',
    },
    bidUsers: [
      {
        id: 10,
        name: 'Martin Acosta',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/07_xrihyb.png',
      },
      {
        id: 11,
        name: 'Antonella Richieri',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/08_fsbhry.png',
      },
    ],
  },
  {
    id: 13,
    instantPrice: '10.05 ETH',
    highestBid: '1.500 ETH',
    author: 'Santiago Echazu',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/01_qgg62f.png',
    type: 'Art',
    stock: 1,
    likes: 30,
    createdAt: '2022-06-10T11:42:33.000Z',
    endsAt: '2022-08-06T11:42:33.000Z',
    media: {
      id: 7,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121542/NFPAISANOS/img_101_y2iczn.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1500/v1657121542/NFPAISANOS/img_101_y2iczn.png',
    },
    attributes: {
      id: 7,
      color: 'orange',
      type: 'legendary',
    },
    bidUsers: [
      {
        id: 12,
        name: 'Martin Acosta',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/07_xrihyb.png',
      },
      {
        id: 13,
        name: 'Antonella Richieri',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/08_fsbhry.png',
      },
      {
        id: 14,
        name: 'Antonella Richieri',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/08_fsbhry.png',
      },
      {
        id: 15,
        name: 'Martin Acosta',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/07_xrihyb.png',
      },
    ],
  },
  {
    id: 14,
    instantPrice: '1.05 ETH',
    highestBid: '0.40 ETH',
    author: 'Alvaro Echazu',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/02_nqwu5e.png',
    type: 'Art',
    stock: 1,
    likes: 400,
    createdAt: '2022-06-15T11:42:33.000Z',
    endsAt: '2022-08-06T11:42:33.000Z',
    media: {
      id: 8,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121536/NFPAISANOS/img_102_a9rkb7.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121536/NFPAISANOS/img_102_a9rkb7.png',
    },
    attributes: {
      id: 8,
      color: 'black',
      type: 'legendary',
    },
    bidUsers: [
      {
        id: 16,
        name: 'Akira Kaneto',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/04_uunxgl.png',
      },
      {
        id: 17,
        name: 'Facundo Aguinaga',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/04_uunxgl.png',
      },
    ],
  },
  {
    id: 15,
    instantPrice: '1.30 ETH',
    highestBid: '0.022 ETH',
    author: 'Sofia Coolen',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/03_ifygi0.png',
    type: 'Photography',
    stock: 2,
    likes: 14,
    createdAt: '2022-06-20T11:42:33.000Z',
    endsAt: '2022-08-06T11:42:33.000Z',
    media: {
      id: 9,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121534/NFPAISANOS/img_115_lolex4.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1500/v1657121534/NFPAISANOS/img_115_lolex4.png',
    },
    attributes: {
      id: 9,
      color: 'blue',
      type: 'uncommon',
    },
    bidUsers: [
      {
        id: 18,
        name: 'Julian Krupka',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/01_qgg62f.png',
      },
      {
        id: 19,
        name: 'Julian Alvarez',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/04_uunxgl.png',
      },
    ],
  },
  {
    id: 16,
    instantPrice: '6.52 ETH',
    highestBid: '0.031 ETH',
    author: 'Flora Quiroga',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/08_fsbhry.png',
    type: 'Photography',
    stock: 1,
    likes: 201,
    createdAt: '2022-06-30T11:42:33.000Z',
    endsAt: '2022-08-06T11:42:33.000Z',
    media: {
      id: 10,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121534/NFPAISANOS/img_119_o3ouhw.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121534/NFPAISANOS/img_119_o3ouhw.png',
    },
    attributes: {
      id: 10,
      color: 'pink',
      type: 'legendary',
    },
    bidUsers: [
      {
        id: 20,
        name: 'Manuel Torres',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/04_uunxgl.png',
      },
      {
        id: 21,
        name: 'Franco Kaus',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/02_nqwu5e.png',
      },
    ],
  },
  {
    id: 17,
    instantPrice: '0.002 ETH',
    highestBid: '0.00005 ETH',
    author: 'Mar√≠a Del Mar Gestoso',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/03_ifygi0.png',
    type: 'Art',
    stock: 1,
    likes: 0,
    createdAt: '2022-06-01T11:10:33.000Z',
    endsAt: '2022-09-06T11:42:33.000Z',
    media: {
      id: 11,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121534/NFPAISANOS/img_133_slwzia.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1500/v1657121534/NFPAISANOS/img_133_slwzia.png',
    },
    attributes: {
      id: 11,
      color: 'red',
      type: 'uncommon',
    },
    bidUsers: [],
  },
  {
    id: 18,
    instantPrice: '0.028 ETH',
    highestBid: '0.00052 ETH',
    author: 'Julian Cerino',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/01_qgg62f.png',
    type: 'Art',
    stock: 10,
    likes: 5,
    createdAt: '2022-06-14T11:10:33.000Z',
    endsAt: '2022-08-06T11:42:33.000Z',
    media: {
      id: 12,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121534/NFPAISANOS/img_114_d5usms.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1500/v1657121534/NFPAISANOS/img_114_d5usms.png',
    },
    attributes: {
      id: 12,
      color: 'black',
      type: 'rare',
    },
    bidUsers: [
      {
        id: 22,
        name: 'Leo Suarez',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/07_xrihyb.png',
      },
    ],
  },
  {
    id: 19,
    instantPrice: '1.00 ETH',
    highestBid: '0.012 ETH',
    author: 'Martin Russo',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/06_mjddoi.png',
    type: 'Art',
    stock: 1,
    likes: 30,
    createdAt: '2022-06-10T11:10:33.000Z',
    endsAt: '2022-07-06T11:42:33.000Z',
    media: {
      id: 13,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121532/NFPAISANOS/img_122_s2bk3s.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1500/v1657121532/NFPAISANOS/img_122_s2bk3s.png',
    },
    attributes: {
      id: 13,
      color: 'black',
      type: 'legendary',
    },
    bidUsers: [
      {
        id: 23,
        name: 'Juan Lazo',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/07_xrihyb.png',
      },
      {
        id: 24,
        name: 'Exequiel Germano',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/02_nqwu5e.png',
      },
    ],
  },
  {
    id: 20,
    instantPrice: '3.00 ETH',
    highestBid: '0.21 ETH',
    author: 'Belen Lago',
    authorAvatar:
      'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/05_rdi4su.png',
    type: 'Photography',
    stock: 5,
    likes: 15,
    createdAt: '2022-06-03T11:10:33.000Z',
    endsAt: '2022-07-06T11:42:33.000Z',
    media: {
      id: 14,
      image:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1000/v1657121534/NFPAISANOS/img_104_zklzrv.png',
      image2x:
        'https://res.cloudinary.com/dvmll0ruo/image/upload/c_scale,w_1500/v1657121534/NFPAISANOS/img_104_zklzrv.png',
    },
    attributes: {
      id: 14,
      color: 'orange',
      type: 'rare',
    },
    bidUsers: [
      {
        id: 25,
        name: 'Nacho Margulies',
        avatar:
          'https://res.cloudinary.com/dvmll0ruo/image/upload/v1657153929/NFPAISANOS/07_xrihyb.png',
      },
    ],
  },
]
