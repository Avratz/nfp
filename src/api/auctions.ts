import { getCachedOneMinute } from '@/lib/fetcher'
import { Auction } from '@/types/auctions'

export function getPopularAuctions(): Promise<Auction[]> {
  return getCachedOneMinute('/popular', process.env.API_KEY as string)
}

export function getAllAuctions(): Promise<Auction[]> {
  return getCachedOneMinute('/aunctions', process.env.API_KEY as string)
}
