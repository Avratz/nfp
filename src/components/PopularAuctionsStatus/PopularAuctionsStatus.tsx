import { getEthereumPrice } from '@/api/coinPrices'
import { getTimeLeft } from '@/lib/time'
import { Auction } from '@/types/auctions'
import { Heading } from '@/ui/Heading/Heading'
import { useQuery } from '@tanstack/react-query'

import styles from './PopularAuctionsStatus.module.css'

export function PopularAuctionsStatus({
  selectedSlide,
}: {
  selectedSlide: Auction
}) {
  const { hours, minutes, seconds } = getTimeLeft(selectedSlide.endsAt)
  const results = useQuery(['eth-price', 'eth'], getEthereumPrice)

  return (
    <div className={styles.auctionStatus}>
      <div>
        <div className={styles.auctionStatusTitle}>Current Bid</div>
        <Heading as="h4" size="5xl">
          {selectedSlide.highestBid}
        </Heading>
        {results.isLoading ? (
          <div className={styles.skeleton}></div>
        ) : results.isError ? (
          <div>C-3PO says he is terribly sorry about all this.</div>
        ) : (
          <Heading as="h5" size="2xl" className={styles.priceUsd}>
            $ {fromEthToUsd(selectedSlide.highestBid, results.data.usd)}
          </Heading>
        )}
      </div>
      <div>
        <div className={styles.auctionStatusTitle}>Auction ending in</div>
        <div className={styles.timer}>
          <div className={styles.time}>
            <div className={styles.timeValue}>{hours}</div>
            <div className={styles.timeUnit}>Hrs</div>
          </div>
          <div className={styles.time}>
            <div className={styles.timeValue}>{minutes}</div>
            <div className={styles.timeUnit}>mins</div>
          </div>
          <div className={styles.time}>
            <div className={styles.timeValue}>{seconds}</div>
            <div className={styles.timeUnit}>secs</div>
          </div>
        </div>
      </div>
    </div>
  )
}

function fromEthToUsd(eth: string, usdOneEth: number) {
  // eth is formatted as 'xxxx ETH'
  let ethNumber = Number(eth.split(' ')[0])

  let total = ethNumber * usdOneEth
  return total.toFixed(2)
}
