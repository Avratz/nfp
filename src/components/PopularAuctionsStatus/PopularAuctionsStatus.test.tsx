import { render, cleanup } from '@testing-library/react'
import { PopularAuctionsStatus } from './PopularAuctionsStatus'
import { Auction } from '@/types/auctions'

jest.mock('@/api/coinPrices', () => ({
  getEthereumPrice: jest.fn(() => Promise.resolve({ usd: 200 })),
}))
jest.mock('@/lib/time', () => ({
  getTimeLeft: jest.fn(() => ({ hours: 1, minutes: 2, seconds: 3 })),
}))
jest.mock('@tanstack/react-query', () => ({
  useQuery: jest.fn(() => ({
    isLoading: false,
    isError: false,
    data: {
      usd: 200,
    },
  })),
}))

describe('PopularAuctionsStatus', () => {
  afterEach(() => {
    cleanup()
    jest.resetAllMocks()
  })

  it('should render the component', () => {
    const selectedSlide = {
      highestBid: '1 ETH',
      endsAt: new Date(),
    } as unknown as Auction

    const { container } = render(
      <PopularAuctionsStatus selectedSlide={selectedSlide} />,
    )
    expect(container).toMatchSnapshot()
  })
})
