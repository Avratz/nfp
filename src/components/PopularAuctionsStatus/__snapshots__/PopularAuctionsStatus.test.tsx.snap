// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularAuctionsStatus should render the component 1`] = `
<div>
  <div
    class="auctionStatus"
  >
    <div>
      <div
        class="auctionStatusTitle"
      >
        Current Bid
      </div>
      <h4
        class="heading 5xl"
      >
        1 ETH
      </h4>
      <h5
        class="heading 2xl priceUsd"
      >
        $ 
        200.00
      </h5>
    </div>
    <div>
      <div
        class="auctionStatusTitle"
      >
        Auction ending in
      </div>
      <div
        class="timer"
      >
        <div
          class="time"
        >
          <div
            class="timeValue"
          >
            1
          </div>
          <div
            class="timeUnit"
          >
            Hrs
          </div>
        </div>
        <div
          class="time"
        >
          <div
            class="timeValue"
          >
            2
          </div>
          <div
            class="timeUnit"
          >
            mins
          </div>
        </div>
        <div
          class="time"
        >
          <div
            class="timeValue"
          >
            3
          </div>
          <div
            class="timeUnit"
          >
            secs
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
